homeassistant/components/local_todo/store.py

```diff
@@ -0,0 +1,38 @@
+"""Local storage for the Local To-do integration."""
+
+import asyncio
+from pathlib import Path
+
+from homeassistant.core import HomeAssistant
+
+STORAGE_PATH = ".storage/{key}.ics"
+
+
+class LocalTodoListStore:
+    """Local storage for a single To-do list."""
+
+    def __init__(self, hass: HomeAssistant, path: Path) -> None:
+        """Initialize LocalTodoListStore."""
+        self._hass = hass
+        self._path = path
+        self._lock = asyncio.Lock()
+
+    async def async_load(self) -> str:
+        """Load the calendar from disk."""
+        async with self._lock:
+            return await self._hass.async_add_executor_job(self._load)
+
```

Comment:
We could use a decorator for this and `async_store` proxy functions

Reply:
Can you elabroate? I'm not familiar with the pattern you are referring to.

Reply:
I think we can address this in a follow up, when we know more.